### PR TITLE
(#22) Add Elixir syntax support

### PR DESCRIPTION
**Pull request by gomoripeti**

### CODE REVIEW

File: priv/test/utils.test.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
+// eslint-env: mocha
+
+import { expect } from "chai";
+
+import Utils from "../app/utils.js";
+
```
Comment (afronski) [inline priv/test/utils.test.js]: *Nitpicking*: Superfluous empty line.

File: priv/test/utils.test.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  describe("Weird characters", function() {
+    it("a0@a:'Abc'/0", function() {
+      expect(Utils.chartId([ "a0@a", "Abc", 0, "a0@a:'Abc'/0" ])).to.be.equal("chart_a0-a--Abc--0");
+    });
+
+    it("'Erlang $!':'is_fun!'/1", function() {
+      expect(Utils.chartId([ "Erlang $!", "is_fun!", 1, "'Erlang $!':'is_fun!'/1" ])).to.be.equal("chart_-Erlang------is_fun---1");
+    });
+  });
+
```
Comment (afronski) [inline priv/test/utils.test.js]: *Nitpicking*: Superfluous empty line.

File: src/xprof_elixir_syntax.erl [hunk 3]
```diff
@@ -0,0 +1,273 @@
+%%%
+%%% @doc Module to parse and format expressions in Elixir syntax
+%%%
+-module(xprof_elixir_syntax).
+
+-behaviour(xprof_language).
+
+-export([parse_query/1,
+
```
Comment (afronski) [inline src/xprof_elixir_syntax.erl]: *Nitpicking*: Superfluous empty line.

File: src/xprof_elixir_syntax.erl [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+-spec quoted_to_ast(ex_quoted()) -> erl_ast().
+quoted_to_ast(Quoted) ->
+    %% pretend that the quoted expression is within a function,
+    %% this way local function calls are allowed
+    %% (also expressions which look like local function calls
+    %%  like 'return_trace()' required for fun2ms)
+    Env = maps:put(function, {ms, 0}, elixir:env_for_eval([])),
+    {Ast, _NewEnv, _Scope} = elixir:quoted_to_erl(Quoted, Env),
+    Ast.
+
```
Comment (afronski) [inline src/xprof_elixir_syntax.erl]: *Nitpicking*: Superfluous empty line.

File: src/xprof_elixir_syntax.erl [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+    case xprof_lib:prefix(Query, <<"Elixir.">>) of
+        true ->
+            %% Query itself is a prefix of "Elixir." - do not prepend
+            Query;
+        _ ->
+            <<"Elixir.", Query/binary>>
+    end;
+normalise_query(Query) ->
+    Query.
+
```
Comment (afronski) [inline src/xprof_elixir_syntax.erl]: *Nitpicking*: Superfluous empty line.

File: src/xprof_erlang_syntax.erl [hunk 6]
```diff
@@ -0,0 +1,104 @@
+%%%
+%%% @doc Module to parse and format expressions in Erlang syntax
+%%%
+-module(xprof_erlang_syntax).
+
+-behaviour(xprof_language).
+
+-export([parse_query/1,
+
```
Comment (afronski) [inline src/xprof_erlang_syntax.erl]: *Nitpicking*: Superfluous empty line.

File: src/xprof_erlang_syntax.erl [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+%% of the clauses of the anonimous function.
+parse_query(Str) ->
+    case tokens(Str) of
+        {mfa, _M, _F, _Arity} = MFA ->
+            MFA;
+        {clauses, M, F, Tokens} ->
+            Clauses = parse(Tokens),
+            {clauses, M, F, Clauses}
+    end.
+
```
Comment (afronski) [inline src/xprof_erlang_syntax.erl]: *Nitpicking*: Superfluous empty line.

File: src/xprof_vm_info.erl [hunk 8]
```diff
@@ -11,100 +11,85 @@
...
+                   || {Fun, Arity} <- filter_funs(FunPrefix, Funs, ModeCB)]
+          end, ExactMatch),
+
+    %% find modules which query is a partial prefix of
+    %% and return only their global/public functions
+    MatchingMods = filter_mods(NormQuery, AllMods, ModeCB),
+    GlobalFuns =
+        [{Mod, Fun, Arity}
+         || Mod <- MatchingMods -- ExactMods,
+            {Fun, Arity} <- get_global_functions(Mod, ModeCB)],
```
Comment (ppikula) [inline src/xprof_vm_info.erl]: maybe all instead of only exported ones?

File: priv/app/function_browser.jsx [hunk 9]
```diff
@@ -41,7 +41,7 @@ class ACModal extends React.Component {
   }
 
   handleFunClick(MFA, e) {
-    var query = Utils.formatMFA(MFA);
+    var query = MFA;
```
Comment (gomoripeti) [inline priv/app/function_browser.jsx]: great idea :)

File: src/xprof_elixir_syntax.erl (hunk 3 above)
Comment (gomoripeti) [inline src/xprof_elixir_syntax.erl]: I just wanted to emphasize division between callbacks for query parsing and callbacks for auto-compete. But probably there are other sublte signs to divide them anyway.

File: src/xprof_vm_info.erl (hunk 8 above)
Comment (gomoripeti) [inline src/xprof_vm_info.erl]: let's address this in a separate PR

Comment (ppikula) [inline src/xprof_vm_info.erl]: OK

**Pull request merged**